### PR TITLE
FIX searchèuser filter expedition list #14290

### PR DESCRIPTION
**Pull request by atm-lena**

### CODE REVIEW

File: htdocs/expedition/list.php [hunk 1]
```diff
@@ -249,8 +249,8 @@
 if ($search_sale > 0 || (!$user->rights->societe->client->voir && !$socid)) $sql .= ", ".MAIN_DB_PREFIX."societe_commerciaux as sc";
 if ($search_user > 0)
 {
-	$sql .= ", ".MAIN_DB_PREFIX."element_contact as ec";
-	$sql .= ", ".MAIN_DB_PREFIX."c_type_contact as tc";
+	$sql .= " LEFT JOIN  ".MAIN_DB_PREFIX."element_contact as ec ON u.rowid = ec.element_id";
```
Comment (eldy) [inline htdocs/expedition/list.php:252]: Note sure this is a good fix.
I did not find "search_user" at all, any where in the where.
So i think we missed instead a criteria where on search_user somewhere.

Comment (atm-lena) [reply htdocs/expedition/list.php:252]: I added "search_user" in where condition.

Comment (eldy) [reply htdocs/expedition/list.php:252]: So there is no reason to use left join on element_contact. Exact join (inner join) should be better for performance when search_user is set.
Also I did not find any use of table alias "tc", so left join on this table may be removed, don't you think so ?

Comment (atm-lena) [reply htdocs/expedition/list.php]: I agree and it’s done !

File: htdocs/expedition/list.php [hunk 2]
```diff
@@ -246,11 +246,10 @@
 $sql .= ' LEFT JOIN '.MAIN_DB_PREFIX.'user as u ON e.fk_user_author = u.rowid';
 
 // We'll need this table joined to the select in order to filter by sale
-if ($search_sale > 0 || (!$user->rights->societe->client->voir && !$socid)) $sql .= ", ".MAIN_DB_PREFIX."societe_commerciaux as sc";
+if ($search_sale > 0 || (!$user->rights->societe->client->voir && !$socid)) $sql .= " LEFT JOIN ".MAIN_DB_PREFIX."societe_commerciaux as sc ON s.rowid = sc.fk_soc";
 if ($search_user > 0)
 {
-	$sql .= ", ".MAIN_DB_PREFIX."element_contact as ec";
-	$sql .= ", ".MAIN_DB_PREFIX."c_type_contact as tc";
+	$sql .= " INNER JOIN  ".MAIN_DB_PREFIX."element_contact as ec ON u.rowid = ec.element_id";
```
Comment (eldy) [inline htdocs/expedition/list.php:252]: The table u is the table to get name of the user fk_user_author.
But the search on user is to find object linked with element_id so we need a link between
ec.element_id and e.rowid

**Pull request closed without merging**